### PR TITLE
Add IModelsGenerator section to MB extending article

### PR DESCRIPTION
**Pull request by sofietoft**

### CODE REVIEW

File: 11/umbraco-cms/reference/templating/modelsbuilder/understand-and-extend.md [hunk 1]
```diff
@@ -117,9 +117,15 @@ Furthermore a generated model will always be instantiated with its default const
 
 For more complex partial classes, you'll have to use the full version of the [Models Builder](https://github.com/zpqrtbnk/Zbu.ModelsBuilder).
 
+### IModelsGenerator
+
+As of Umbraco 11.4, the IModelsGenerator interface has been added. The interface manipulates how to Models Builder works, allowing you to extend the different features.
+
+The interface can be accessed via `Infrastructure.ModelsBuilder.Building.ModelsGenerator`.
```
Comment (Zeegaan) [inline 11/umbraco-cms/reference/templating/modelsbuilder/understand-and-extend.md:124]:
```suggestion
As of Umbraco 11.4, the IModelsGenerator interface has been added. If you want to customize how the models are generated, you can make your own implementation of the `IModelsGenerator` interface, and then overwrite the Umbraco implementation with dependency injection. 

The interface can be accessed via `Infrastructure.ModelsBuilder.Building.ModelsGenerator`.
```

**Review by Zeegaan — CHANGES_REQUESTED** [review]
I've changed the wording a bit, to make it more clear 😁

File: 11/umbraco-cms/reference/templating/modelsbuilder/understand-and-extend.md [hunk 2]
```diff
@@ -117,9 +117,15 @@ Furthermore a generated model will always be instantiated with its default const
 
 For more complex partial classes, you'll have to use the full version of the [Models Builder](https://github.com/zpqrtbnk/Zbu.ModelsBuilder).
 
+### IModelsGenerator
+
+As of Umbraco 11.4, the IModelsGenerator interface has been added. If you want to customize how the models are generated, you can make your own implementation of the `IModelsGenerator` interface, and then overwrite the Umbraco implementation with dependency injection.
```
Comment (sofietoft) [inline 11/umbraco-cms/reference/templating/modelsbuilder/understand-and-extend.md:122]: I've committed your suggestion @Zeegaan , but only to make another suggestion to that suggestion (🙃 ), to shorten the sentence a bit!
Let me know if it still works 🤞 

```suggestion
As of Umbraco 11.4, the IModelsGenerator interface has been added. If you want to customize how the models are generated, you can make your own implementation of the `IModelsGenerator` interface. You can then overwrite the Umbraco implementation with dependency injection. 
```

Comment (Zeegaan) [reply 11/umbraco-cms/reference/templating/modelsbuilder/understand-and-extend.md:122]: 🤣 
Looks good 💪

**Review by Zeegaan — APPROVED** [review]

**Pull request merged**